lib/pure/asyncfile.nim

```diff
@@ -81,11 +81,10 @@ proc getFileSize*(f: AsyncFile): int64 =
   else:
     result = lseek(f.fd.cint, 0, SEEK_END)
 
-proc newAsyncFile*(fd: AsyncFd): AsyncFile =
+proc newAsyncFile*(fd: cint | AsyncFd): AsyncFile =
   ## Creates `AsyncFile` with a previously opened file descriptor `fd`.
   new result
-  result.fd = fd
-  register(result.fd)
+  result.fd = register(fd)
```

Comment:
Confusing chaining.

Reply:
You can think of this as: ``register`` takes an FD and returns a "registered" instance of that FD (cint -> AsyncFD in this case).

Reply:
Yes, I do undestand the code. But I dislike its design.